app/src/main/java/com/mk/tasky/agenda/domain/usecase/task/DeleteTask.kt

```diff
@@ -0,0 +1,13 @@
+package com.mk.tasky.agenda.domain.usecase.task
+
+import com.mk.tasky.agenda.domain.repository.AgendaRepository
+
+class DeleteTask(
+    private val repository: AgendaRepository
+) {
+    suspend operator fun invoke(id: String) {
+        repository.deleteTaskById(id).onFailure {
+            // TODO: Save id on db to later sync with server
```

Comment:
This is now debatable, but if the caching strategy is implemented in the repo, I'd also leave it up to the repo when you want save something to sync later since that's part of the caching strategy.
Some people consider a caching strategy as business logic and implement it on domain, then it would be fine to do it in the use case, but I think it should be consistent.

Reply:
Sounds good. Will make it on the repository!